demisto_sdk/commands/content_graph/parsers/related_files.py

```diff
@@ -0,0 +1,280 @@
+from abc import ABC
+from enum import Enum
+from pathlib import Path
+from typing import Any, ClassVar, List, Optional, Union
+
+from demisto_sdk.commands.common.constants import (
+    AUTHOR_IMAGE_FILE_NAME,
+    PACKS_PACK_IGNORE_FILE_NAME,
+    PACKS_README_FILE_NAME,
+    PACKS_WHITELIST_FILE_NAME,
+    RELEASE_NOTES_DIR,
+    GitStatuses,
+)
+from demisto_sdk.commands.common.files import TextFile
+from demisto_sdk.commands.common.git_util import GitUtil
+from demisto_sdk.commands.common.logger import logger
+
+
+class RelatedFileType(Enum):
+    YML = "yml"
+    JSON = "json"
+    README = "readme"
+    DESCRIPTION_File = "description_file"
+    IMAGE = "image"
+    DARK_SVG = "dark_svg"
+    LIGHT_SVG = "light_svg"
+    CODE_FILE = "code_file"
+    TEST_CODE_FILE = "test_code_file"
+    SCHEMA = "schema_file"
+    XIF = "xif_file"
+    PACK_IGNORE = "pack_ignore"
+    SECRETS_IGNORE = "secrets_ignore"
+    AUTHOR_IMAGE = "author_image_file"
+    RELEASE_NOTE = "release_note"
+
+
+class RelatedFile(ABC):
+    file_type: ClassVar[RelatedFileType]
+
+    def __init__(self, main_file_path: Path, git_sha: Optional[str] = None) -> None:
+        self.main_file_path: Path = main_file_path
+        self.git_sha = git_sha
+        self.exist: bool = False
+        self.file_path: Path = self.find_the_right_path(self.get_optional_paths())
+        self.git_status: Union[GitStatuses, None] = self.get_git_status()
+
+    def get_git_status(self) -> Union[GitStatuses, None]:
+        status = None
+        if self.git_sha:
+            git_util = GitUtil.from_content_path()
+            remote, branch = git_util.handle_prev_ver(
+                self.git_sha  # type: ignore[arg-type]
```

Comment:
```suggestion
                self.git_sha
```

cast the var to avoid mypy ignores?

Reply:
it's wrapped by an if statement so it should be alright.